naloge/bfs.tex

```diff
@@ -10,6 +11,112 @@
 \caption{Graf za nalogi~\nal in~\nal[dfs].}
 \end{slika}
 \end{vprasanje}
+
 \begin{odgovor}
+ALGORITEM:\\
+\underline{VHOD}:  graf G, vozlišče $s \in V$\\
+\underline{IZHOD}: razdalje $d_G(s, v)$ za vsak $v \in V$\\
+def bfs(G, s):\\
+	\phantom{x}\hspace{3ex}za vsak $v \in V(G)$:\\
+		\phantom{x}\hspace{6ex}razdalja[v] = $\infty$\\
+	\phantom{x}\hspace{3ex}razdalja[s] = 0\\
+	\phantom{x}\hspace{3ex}Q = [s]\\
+	\phantom{x}\hspace{3ex}dokler Q ni prazna:\\
+		\phantom{x}\hspace{6ex}u = odstrani-prvega(Q)\\
+		\phantom{x}\hspace{6ex}za vsak $(u, v) \in E(G)$:\\
+			\phantom{x}\hspace{9ex}če razdalja[v] = $\infty$:\\
+				\phantom{x}\hspace{12ex}razdalja[v] = razdalja[u] + 1\\
+				\phantom{x}\hspace{12ex}vstavi(Q, v)\\
+\begin{flushleft}
+Če sledimo zgornjemu algoritmu in predpostavki, da upoštevamo vrstni red, začnemo v vozlišču A, je potek sledeč: 
+\end{flushleft}
+\underline{Predpriprava:}\\
```

Comment:
Morda bo bolj berljivo in kompaktno, če izvajanje algoritma predstaviš v tabeli, tako kot npr. v [`bubblesort.tex`](https://github.com/jaanos/or-zbirka/blob/master/naloge/bubblesort.tex).

Reply:
Sem si ogledala primer in ne vem točno, na kakšen način naj bi naredila tabelo za algoritem bfs, saj je več zank in ne le ena. Po kateri zanki naj naredim tabelo?

Reply:
Tabelo narediš na podoben primer: v enem stolpcu naj bo vrednost *s* (ki bo konstantna v obhodu zunanje zanke **for**), v naslednjih vrednosti *u* in *v*, potem pa še vsebina *Q* in množica označenih vozlišč (bo to bolj pregledno kot seznam vrednosti `\True` in `\False`). Vsakemu trenutnemu koraku bo torej ustrezala ena ali dve vrstici.